voidcrew/modules/cloning/code/computer_cloning.dm

```diff
@@ -0,0 +1,526 @@
+	say("Initiating scan...")
+	var/prev_locked = scanner.locked
+	scanner.locked = TRUE
+	addtimer(CALLBACK(src, PROC_REF(finish_scan), scanner.occupant, user, prev_locked, body_only), 2 SECONDS)
+	return TRUE
+
+/obj/machinery/computer/cloning/proc/toggle_autoprocess(mob/user)
+	if(!scanner || !HasEfficientPod() || scanner.scan_level < AUTOCLONING_MINIMAL_LEVEL)
+		return FALSE
+	autoprocess = !autoprocess
+	if(autoprocess)
+		START_PROCESSING(SSmachines, src)
+		playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+	else
+		STOP_PROCESSING(SSmachines, src)
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	return TRUE
+
+/obj/machinery/computer/cloning/proc/finish_scan(mob/living/L, mob/user, prev_locked, body_only)
+	if(!scanner || !L)
+		return
+	src.add_fingerprint(usr)
+	scan_occupant(L, user, body_only)
+
+	loading = FALSE
+	scanner.locked = prev_locked
+	playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+	SStgui.update_uis(src) // Immediate since it's not spammable
+
+//Used by consoles without records
+/obj/machinery/computer/cloning/proc/clone_occupant(occupant, mob/user)
+	var/mob/living/mob_occupant = get_mob_or_brainmob(occupant)
+	var/datum/dna/dna
+	if(ishuman(mob_occupant))
+		var/mob/living/carbon/C = mob_occupant
+		dna = C.has_dna()
+	if(isbrain(mob_occupant))
+		var/mob/living/brain/B = mob_occupant
+		dna = B.stored_dna
+	if(!can_scan(dna, mob_occupant))
+		return
+	var/clone_species
+	if(dna.species)
+		clone_species = dna.species
+	else
+		scantemp = "Unauthorized clone process detected => Interrupted."
+		return //no dna info for species? you're not allowed to clone them. Don't harass xeno, don't try xeno farm.
+	var/obj/machinery/clonepod/pod = GetAvailablePod()
+	//Can't clone without someone to clone.  Or a pod.  Or if the pod is busy. Or full of gibs.
+	if(!LAZYLEN(pods))
+		temp = "No Clonepods detected."
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else if(!pod)
+		temp = "No Clonepods available."
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else if(pod.occupant)
+		temp = "Cloning cycle already in progress."
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else
+		pod.growclone(mob_occupant.real_name, dna.unique_identity, dna.mutation_index, null, null, clone_species, dna.blood_type, mob_occupant.faction)
+		temp = "[mob_occupant.real_name] => Cloning data sent to pod."
+		playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+		log_combat("[user ? key_name(user) : "Unknown"] cloned [key_name(mob_occupant)] with [src] at [AREACOORD(src)].")
+
+/obj/machinery/computer/cloning/proc/can_scan(datum/dna/dna, mob/living/mob_occupant, datum/bank_account/account, body_only)
+	if(!istype(dna))
+		scantemp = "Unable to locate valid genetic data."
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+		return FALSE
+	if(NO_DNA_COPY in dna.species.species_traits)
```

Comment:
is this still functional?

Reply:
So It seems i used the wrong code once more... I will go fix up the code this was to use

Reply:
I don't think so